cfme/tests/cloud/test_project_quota.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from widgetastic.utils import partial_match
+
+from cfme import test_requirements
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.generators import random_vm_name
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.long_running,
+    pytest.mark.provider([OpenStackProvider], required_fields=[['provisioning', 'image']],
+                         scope="module")
+]
+
+
+@pytest.fixture
+def vm_name():
+    return random_vm_name(context='quota')
+
+
+@pytest.fixture
+def template_name(provisioning):
+    return provisioning["image"]["name"]
+
+
+@pytest.fixture(scope="module")
+def roottenant(appliance):
+    return appliance.collections.tenants.get_root_tenant()
+
+
+@pytest.fixture
+def prov_data(vm_name, template_name):
+    return {
+        "catalog": {'vm_name': vm_name, 'catalog_name': {'name': template_name}},
+        "environment": {'automatic_placement': True},
+        'properties': {'instance_type': partial_match('m1.large')}
+    }
+
+
+@pytest.fixture
+def custom_prov_data(request, prov_data, vm_name, template_name):
+    value = request.param
+    prov_data.update(value)
+    prov_data['catalog']['vm_name'] = vm_name
+    prov_data['catalog']['catalog_name'] = {'name': template_name}
+
+
+@pytest.fixture
+def set_project_quota(request, appliance, new_project):
+    field, value = request.param
+    new_project.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
+    new_project.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope='module')
+def new_project(appliance):
+    collection = appliance.collections.projects
+    yield collection.create(name='project{}'.format(fauxfactory.gen_alphanumeric()),
+                            description='project_des{}'.format(fauxfactory.gen_alphanumeric()),
+                            parent=collection.get_root_tenant())
+
+
+@pytest.yield_fixture(scope='module')
```

Comment:
Required: yield_fixture is not mandatory to use now as @pytest.fixture in new version of pytest do the same